platform/contracts/admin/src/contracts/protocol/mod.rs

```diff
@@ -22,5 +22,7 @@ pub struct ProtocolTemplate<T> {
 #[serde(rename_all = "snake_case", deny_unknown_fields)]
 pub struct Protocol {
     pub network: String,
+    pub dex_type: String,
```

Comment:
using `String`s here is not adequate .... we have to restrict to the supported ones